docs/docs/how-tos/wait-user-input-functional.ipynb

```diff
@@ -0,0 +1,569 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _print_step(step: dict) -> None:\n",
+    "    for task_name, result in step.items():\n",
+    "        if task_name in (\"agent\", \"__interrupt__\"):\n",
+    "            continue  # Just print task updates\n",
+    "        else:\n",
+    "            print(f\"\\n{task_name}:\")\n",
+    "            result.pretty_print()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config = {\"configurable\": {\"thread_id\": \"1\"}}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'role': 'user', 'content': 'Can you reach out for human assistance: what should I feed my cat? Separately, can you check the weather in San Francisco?'}\n",
+      "\n",
+      "call_model:\n",
+      "==================================\u001b[1m Ai Message \u001b[0m==================================\n",
+      "Tool Calls:\n",
+      "  human_assistance (call_Er3DsMcpdoO9yRy2NZHXV6mH)\n",
+      " Call ID: call_Er3DsMcpdoO9yRy2NZHXV6mH\n",
+      "  Args:\n",
+      "    query: What should I feed my cat?\n",
+      "  get_weather (call_vkBLyCTkmGR2vMndJyxnfzn7)\n",
+      " Call ID: call_vkBLyCTkmGR2vMndJyxnfzn7\n",
+      "  Args:\n",
+      "    location: San Francisco\n",
+      "\n",
+      "call_tool:\n",
+      "=================================\u001b[1m Tool Message \u001b[0m=================================\n",
+      "\n",
+      "It's sunny!\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_message = {\n",
+    "    \"role\": \"user\",\n",
+    "    \"content\": (\n",
+    "        \"Can you reach out for human assistance: what should I feed my cat? \"\n",
+    "        \"Separately, can you check the weather in San Francisco?\"\n",
+    "    ),\n",
+    "}\n",
+    "print(user_message)\n",
+    "\n",
+    "for step in agent.stream([user_message], config):\n",
+    "    _print_step(step)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that we generate two tool calls, and although our run is interrupted, we did not block the execution of the `get_weather` tool.\n",
+    "\n",
+    "Let's inspect where we're interrupted:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(Interrupt(value={'query': 'What should I feed my cat?'}, resumable=True, ns=['agent:5e01f3a3-a097-3cb2-7557-b36fa4c5e427', 'call_tool:154fa07a-05dc-b025-2a4d-e7faa29c643a'], when='during'),)"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "next_task = agent.get_state(config).tasks[0]\n",
```

Comment:
(nit) nothing wrong with showing interrupts this way, but you can also just simply stream updates instead, and interrupt will be streamed as well